kuma/wiki/models.py

```diff
@@ -1292,7 +1293,9 @@ def get_redirect_url(self):
                 # i.e allow "https://developer...." and "/en-US/docs/blah"
                 if len(url) > 1:
                     if url.startswith(settings.SITE_URL):
-                        return url
+                        parsed = urlparse(url)
```

Comment:
It seems to me this is the only change needed to fix the bug, and the change to ``get_doc_components_from_url`` is unneeded. Or, this change is unneeded if you change ``get_doc_components_from_url``...

In either case, this breaks a test, as reported by TravisCI.